src/main/java/pl/debuguj/parkingspacessystem/dao/ParkingSpaceDao.java

```diff
@@ -0,0 +1,29 @@
+package pl.debuguj.parkingspacessystem.dao;
+
+import pl.debuguj.parkingspacessystem.exceptions.CarRegisteredInSystemException;
+import pl.debuguj.parkingspacessystem.exceptions.IncorrectEndDateException;
+import pl.debuguj.parkingspacessystem.domain.ParkingSpace;
+import pl.debuguj.parkingspacessystem.exceptions.ParkingSpaceNotFoundException;
+
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Created by grzesiek on 07.10.17.
+ */
+public interface ParkingSpaceDao {
+
+    void create(final ParkingSpace parkingSpace) throws CarRegisteredInSystemException;
+
+    Optional<ParkingSpace> findByRegistrationNo(final String registrationNo);
+
+    List<ParkingSpace> getAll();
```

Comment:
To brzmi jak niepotrzebne męczenie bazy - do czego to może być wykorzystane? Do sprawdzania potem czy coś jest zajęte? W takim wypadku lepiej to zrobić na bazie

Reply:
Potwierdzam niepotrzebne męczenie bazy danych. :)